File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        mp=0
        mi=prices[0]
        for i in prices:
            if i<mi:
                mi=i
            p=i-mi
            if p>mp:
             mp=p
        return mp
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {

        int max_p = 0;
        int mi = prices[0];

        for (int i = 1; i < prices.length; i++) {

            if (prices[i] < mi) {
                mi = prices[i];
            }

            int profit = prices[i] - mi;

            if (profit > max_p) {
                max_p = profit;
            }
        }

        return max_p;
    }
}